central/auth/datastore/datastore_impl_test.go

```diff
@@ -0,0 +1,315 @@
+		Type:                    storage.AuthMachineToMachineConfig_GITHUB_ACTIONS,
+		TokenExpirationDuration: "5m",
+		Mappings: []*storage.AuthMachineToMachineConfig_Mapping{
+			{
+				Key:   "sub",
+				Value: "some-value",
+				Role:  testRole1,
+			},
+		},
+	})
+
+	s.NoError(err)
+
+	s.Equal("https://token.actions.githubusercontent.com", config.GetIssuer())
+}
+
+func (s *datastorePostgresTestSuite) TestAddGitHubActionConfigCustomIssuer() {
+	config, err := s.authDataStore.AddAuthM2MConfig(s.ctx, &storage.AuthMachineToMachineConfig{
+		Id:                      "80c053c2-24a7-4b97-bd69-85b3a511241e",
+		Type:                    storage.AuthMachineToMachineConfig_GITHUB_ACTIONS,
+		TokenExpirationDuration: "5m",
+		Mappings: []*storage.AuthMachineToMachineConfig_Mapping{
+			{
+				Key:   "sub",
+				Value: "some-value",
+				Role:  testRole1,
+			},
+		},
+		Issuer: "https://stackrox.io",
+	})
+
+	s.NoError(err)
+
+	s.Equal("https://stackrox.io", config.GetIssuer())
+}
+
+func (s *datastorePostgresTestSuite) TestUpdateGitHubActionConfigFillIssuer() {
+	config, err := s.authDataStore.AddAuthM2MConfig(s.ctx, &storage.AuthMachineToMachineConfig{
+		Id:                      "80c053c2-24a7-4b97-bd69-85b3a511241e",
+		Type:                    storage.AuthMachineToMachineConfig_GENERIC,
+		TokenExpirationDuration: "5m",
+		Mappings: []*storage.AuthMachineToMachineConfig_Mapping{
+			{
+				Key:   "sub",
+				Value: "some-value",
+				Role:  testRole1,
+			},
+		},
+		Issuer: "https://stackrox.io",
+	})
+
+	s.NoError(err)
+
+	config.Type = storage.AuthMachineToMachineConfig_GITHUB_ACTIONS
+	config.Issuer = ""
+
+	err = s.authDataStore.UpdateAuthM2MConfig(s.ctx, config)
+	s.NoError(err)
+
+	updatedConfig, exists, err := s.authDataStore.GetAuthM2MConfig(s.ctx, config.GetId())
+	s.NoError(err)
+	s.True(exists)
+	s.Equal("https://token.actions.githubusercontent.com", updatedConfig.GetIssuer())
+}
+
+func (s *datastorePostgresTestSuite) TestUpdateGitHubActionConfigCustomIssuer() {
+	config, err := s.authDataStore.AddAuthM2MConfig(s.ctx, &storage.AuthMachineToMachineConfig{
+		Id:                      "80c053c2-24a7-4b97-bd69-85b3a511241e",
+		Type:                    storage.AuthMachineToMachineConfig_GENERIC,
+		TokenExpirationDuration: "5m",
+		Mappings: []*storage.AuthMachineToMachineConfig_Mapping{
+			{
+				Key:   "sub",
+				Value: "some-value",
+				Role:  testRole1,
+			},
+		},
+		Issuer: "https://stackrox.io",
+	})
+
+	s.NoError(err)
+
+	config.Type = storage.AuthMachineToMachineConfig_GITHUB_ACTIONS
+	config.Issuer = "https://private.stackrox.io"
+
+	err = s.authDataStore.UpdateAuthM2MConfig(s.ctx, config)
+	s.NoError(err)
+
+	updatedConfig, exists, err := s.authDataStore.GetAuthM2MConfig(s.ctx, config.GetId())
+	s.NoError(err)
+	s.True(exists)
+	s.Equal("https://private.stackrox.io", updatedConfig.GetIssuer())
+}
+
+func (s *datastorePostgresTestSuite) TestAddUniqueConstraint() {
```

Comment:
Well it's not true, the ID is a different value for this, but I can make it a little bit more obvious.
I don't think there should be both, since the Upsert call will not error out in case the ID already exist, but rather overwrite things.